screenshots/generate_new_configs.py

```diff
@@ -0,0 +1,94 @@
+import os
+import io
+import pandas as pd
+import requests
+import yaml
+
+def load_urls_from_live_info(url):
+
+    content = requests.get(url).content
+    state_info_df = pd.read_csv(io.StringIO(content.decode('utf-8')), keep_default_na=False)
+    
+    state_urls = {}
+    for idx, r in state_info_df.iterrows():
+        state_urls[r["state"]] = {
+            'primary': r["covid19Site"],
+            'secondary': r["covid19SiteSecondary"],
+            'tertiary': r["covid19SiteTertiary"],
+            'quaternary': r["covid19SiteQuaternary"],
+            'quinary': r["covid19SiteQuinary"],
+        }
+
+    return state_urls
+
+
+def load_urls_from_spreadsheet(csv_url):
+
+    urls_df = pd.read_csv(csv_url, keep_default_na=False)
+
+    state_urls = {}
+    for idx, row in urls_df.iterrows():
+        state_urls[row['State']] = {
+            'primary': row['Link 1'],
+            'secondary': row['Link 2'],
+            'tertiary': row['Link 3'],
+        }
+
+    return state_urls
+
+# construct an array containing the URL configs (up to 5) for each state (including the URL, name and existing YAML config)
+# then dump that dictionary to a new YAML file named after the state, in the appropriate directory
+def output_yamls(team, state_urls, config_basename):
+    
+    screenshot_config_path = os.path.join(os.path.dirname(__file__), 'configs', config_basename)
+    with open(screenshot_config_path) as f:
+        existing_config = yaml.safe_load(f)
+
+    # set up the output directory
+    destination_directory = "./config/" + team + "/"
+    os.makedirs(destination_directory, exist_ok=True)
+
+    # walk through the states from the spreadsheet
+    for state, urls in state_urls.items():
+
+        # open a file for the current state, creating the file if it doesn't exist
+        outfile = open(destination_directory + state + '.yaml', 'w+')
+
+        # walk through the URLs for the state, and merge with config from the old YAMLs
+        new_config_list = []
+        for urlname, urltext in urls.items():
+            
+            # if there's no URL, that means there was an empty cell in the spreadsheet
+            if(urltext):
+                existing_state_config = existing_config[urlname].get(state)
+                
+                outfile.write("- name: " + urlname + "\n")
```

Comment:
this is a small change, but I think we talked about having the state name in the config as well so that it's not just inferred from the filename. So the final YAML would look like this:
```
state: AK
links:
  - name: primary
    url: http://dhss.alaska.gov/dph/Epi/id/Pages/COVID-19/monitoring.aspx
    overseerScript: |
      page.manualWait();  await page.waitForDelay(30000);  page.done();
    message: stuff

  - name: secondary
    url: https://experience.arcgis.com/experience/ed1c874ca60b4c15ab09095a070065ca
...  
```

In terms of using pyyaml vs formatting things manually: From a larger standpoint, I'm all for easier, though having to indent things manually to format them correctly is a little eek-inducing. It seems like there is a way to wrangle pyyaml into formatting multi-line strings (e.g. https://stackoverflow.com/questions/8640959/how-can-i-control-what-scalar-form-pyyaml-uses-for-my-data) but I also totally understand if you don't want to rewrite all this. I can also help with figuring out how to get pyyaml to do what we want?

Reply:
I don't remember that, but I'm cool with it. Otherwise we would infer it from the filename, but that can be problematic, I agree.

As for the formatting, I did some searching and experimenting and couldn't easily get the formatting that I wanted. I'll give the stackoverflow you sent a try, but if it still doesn't quite do it, I'd prefer to stick with manual since this is just a temporary script.

Reply:
I wasn't able to get the control that I was looking for via that slackexchange link, or others that I found. (without switching libraries) I'm inclined to stay with the manual \ning because it allows the most control, and it's pretty simple.